src/Tracer.php

```diff
@@ -165,7 +165,9 @@ private function saveResults(): void
             $this->outFileName,
             \json_encode(
                 $this->result,
-                self::DEBUG_PRETTY_JSON ? JSON_PRETTY_PRINT : 0
+                JSON_INVALID_UTF8_SUBSTITUTE | (
+                    self::DEBUG_PRETTY_JSON ? JSON_PRETTY_PRINT : 0
+                )
```

Comment:
I think it would be easier to read as:


```suggestion
                JSON_INVALID_UTF8_SUBSTITUTE
                | (self::DEBUG_PRETTY_JSON ? JSON_PRETTY_PRINT : 0)
```